libraries/key_vault.rb

```diff
@@ -0,0 +1,55 @@
+# helper for retrieving a secret from an azure keyvault
+module Azure
+  module KeyVault
+    include Azure::Cookbook
+
+    def vault_secret(vault, secret_name, spn = {}, version = nil)
+      request_url = vault_request_url(vault, secret_name, version)
+      token_provider = create_token_provider(spn)
+      headers = {
+        'Authorization' => token_provider.get_authentication_header
+      }
+      http_client = Chef::HTTP.new(request_url, headers)
+      response = http_client.get(http_client.url, headers)
+      JSON.parse(response)['value']
+    end
+
+    private
+
+    def token_audience
+      setup_arm_compute
+      @audience ||= MsRestAzure::ActiveDirectoryServiceSettings.new.tap do |s|
+        s.authentication_endpoint = 'https://login.windows.net/'
+        s.token_audience = 'https://vault.azure.net'
+      end
+    end
+
+    def create_token_provider(spn)
+      validate_service_principle(spn)
+      tenant_id = spn['tenant_id']
+      client_id = spn['client_id']
+      secret = spn['secret']
+      @token_provider ||= begin
+        MsRestAzure::ApplicationTokenProvider.new(tenant_id, client_id, secret, token_audience)
+      end
+    end
+
+    def vault_request_url(vault, secret_name, version, resource = 'secrets', api_version = '2015-06-01')
+      base_url = "https://#{vault}.vault.azure.net/#{resource}/#{secret_name}"
+      version.nil? || base_url << "/#{version}"
+      base_url << "?api-version=#{api_version}"
+      Chef::Log.info(base_url)
```

Comment:
I'd either make this `#debug` or at least prefix some text indicating where it's coming from so the user knows why this is getting logged.